clustered/client/src/main/java/org/ehcache/clustered/client/internal/loaderwriter/writebehind/ClusteredWriteBehindStore.java

```diff
@@ -0,0 +1,385 @@
+                ByteBuffer payload = element.getPayload();
+                Operation<K, V> operation = codec.decode(payload);
+                K key = operation.getKey();
+                PutOperation<K, V> result = resolver.applyOperation(key,
+                                                                    currentState.get(key),
+                                                                    operation,
+                                                                    timeSource.getTimeMillis());
+                try {
+                  if (result != null) {
+                    if (result != currentState.get(key) && !(operation instanceof PutOperation)) {
+                      cacheLoaderWriter.write(result.getKey(), result.getValue());
+                    }
+                    currentState.put(key, result.asOperationExpiringAt(result.expirationTime()));
+                  } else {
+                    if (currentState.get(key) != null && (operation instanceof RemoveOperation
+                                                          || operation instanceof ConditionalRemoveOperation)) {
+                      cacheLoaderWriter.delete(key);
+                    }
+                    currentState.put(key, null);
+                  }
+                } catch (Exception e) {
+                  throw new RuntimeException(e);
+                }
+              }
+
+              ChainBuilder builder = new ChainBuilder();
+              for (PutOperation<K, V> operation : currentState.values()) {
+                builder = builder.add(codec.encode(operation));
+              }
+
+              storeProxy.replaceAtHead(hash, chain, builder.build());
+            }
+          } finally {
+            getProxy().unlock(hash);
+          }
+        } catch (TimeoutException e) {
+          throw new RuntimeException(e);
+        }
+      });
+    }
+  }
+
+  /**
+   * Provider of {@link ClusteredWriteBehindStore} instances.
+   */
+  @ServiceDependencies({ TimeSourceService.class, ClusteringService.class})
+  public static class Provider extends ClusteredLoaderWriterStore.Provider {
+    @Override
+    protected <K, V> ClusteredStore<K, V> createStore(Configuration<K, V> storeConfig,
+                                                      OperationsCodec<K, V> codec,
+                                                      ChainResolver<K, V> resolver,
+                                                      TimeSource timeSource,
+                                                      boolean useLoaderInAtomics,
+                                                      Object[] serviceConfigs) {
+      WriteBehindConfiguration writeBehindConfiguration = findSingletonAmongst(WriteBehindConfiguration.class, serviceConfigs);
+      if (writeBehindConfiguration != null) {
+        ExecutorService executorService =
+          executionService.getOrderedExecutor(writeBehindConfiguration.getThreadPoolAlias(),
```

Comment:
What if write-behind config does not have thread pool provided?
In that case you will have to use defaultThreadPool to get the executor service.

Reply:
ExecutorService implementations provides default thread pool if threadPoolAlias is null